docs/networking/deep-dive.md

```diff
@@ -0,0 +1,102 @@
+
+### Bridge Network
+
+We leverage [multus CNI](https://github.com/k8snetworkplumbingwg/multus-cni) and [bridge CNI](https://www.cni.dev/plugins/current/main/bridge/) to implement the bridge network.
+
+1. Multus CNI is a container network interface (CNI) plugin for Kubernetes that enables attaching multiple network interfaces to pods. Its capability allows a VM to be equipped with both a NIC for the management network and multiple NICs for the bridge network. Without multus CNI, the bridge CNI will not appear in Harvester.
+
+2. With bridge CNI, the VM pods will be plugged into an L2 bridge specified in the network attachment definition config.
+
+    ```json
+    # Example 1
+    {
+        "cniVersion": "0.3.1",
+        "name": "vlan100",
+        "type": "bridge",
+        "bridge": "mgmt-br",
+        "promiscMode": true,
+        "vlan": 100,
+    }
+    ```
+
+    ```json
+    # Example 2
+    {
+        "cniVersion": "0.3.1",
+        "name": "untagged-network",
+        "type": "bridge",
+        "bridge": "oob-br",
+        "promiscMode": true,
+        "ipam": {}
+    }
+    ```
+
+    Example 1 is a typical VLAN configuration whose VLAN ID is 100, while example 2 is an untagged network configuration without a VLAN ID. The VM pods using example 1 configuration will be plugged into the bridge mgmt-br, while those using example 2 will be plugged into the bridge oob-br.
+
+3. For fault tolerance and high availability, the uplink of every bridge is a bond device. As a bridge port, the uplink should allow the tagged package to pass.
+
+    ```shell
+    harvester-0:/home/rancher # bridge -c vlan show dev oob-bo
+    port	   vlan ids
+    oob-bo	   1 PVID Egress Untagged
+               100
+               200
+    ```
+
+    The example above shows that the bond `oob-bo` allows packages with tag 1, 100 or 200.
+
+### Management Network
+
+The management network is based on [canal](https://projectcalico.docs.tigera.io/getting-started/kubernetes/flannel/flannel).
+
+It is worth mentioning that the canal interface is the bridge `mgmt-br` or the VLAN sub-interface of `mgmt-br`, where Harvester configures the node IP. This design has two benefits:
+
+- The built-in cluster network mgmt supports the management network and bridge network simultaneously.
+- Through a VLAN network interface, we can specify a VLAN ID to the management network.
+
+## External Networking
+
+The external network devices usually include switches and a DHCP server. After introducing the cluster network, we can group the host's NICs and connect them to different switches for traffic isolation. Below are some usage notes.
+
+- To allow the passing of the tagged packages, you need to set the type of external switch ports connected to the hosts or other devices (such as the DHCP server) as trunk or hybrid and permit the specified VLANs.
+
+- We need to configure link aggregation on the switch based on the bond mode of the peer host.
```

Comment:
can we add an example of this?

Reply:
List the correspondence between bond mode and link aggregation mode. And we will provide examples in the user case document. cc @tjjh89017